app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

```diff
@@ -186,17 +196,17 @@ public void errorCallback(int errorCode, int numErrors) {
     }
 
     @Override
-    public void commentCallback(String comment) {
+    public void commentCallback(String comment, ArrayList<String> linkInfo) {
         if (comment == null) {
             Log.d(DEBUG_TAG, "No valid comments found");
             Toast.makeText(getContext(), "No valid comments available", Toast.LENGTH_SHORT).show();
         } else {
+            link = "https://www.reddit.com/comments/" + linkInfo.get(0) + "/_/" + linkInfo.get(1) + "?context=2";
```

Comment:
comment why context=2.

Also, we might want to later create this URL using a Builder, but that's not that important as long as this works for all cases.